### PR TITLE
Upgraded dev-deps

### PR DESCRIPTION
**Pull request by GrahamCampbell**
Mimics PHPUnit version on 1.x. Upgrades `http-interop/http-factory-tests`.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -41,8 +41,8 @@
     },
     "require-dev": {
         "bamarni/composer-bin-plugin": "^1.4.1",
-        "http-interop/http-factory-tests": "^0.7",
-        "phpunit/phpunit": "^8.5"
+        "http-interop/http-factory-tests": "^0.9",
+        "phpunit/phpunit": "^8.5.8 || ^9.3.10"
```
Comment (Nyholm) [inline composer.json:45]: Im not a big fan of this line. 

Since we are not using any features in 9.3, there is no need to install it, right?

Comment (GrahamCampbell) [reply composer.json:45]: I think the reason for that line was it is the first version of PHPUnit that is not broken on PHP 8.0.0.

Comment (Nyholm) [reply composer.json:45]: I just think this opens the door to something like this...

https://github.com/guzzle/psr7/blob/1.x/composer.json#L24

Anyhow, lets have a proper discussion about this when someone likes to add a third major version.

**Review by Nyholm — APPROVED** [review]
Thank you

**Pull request merged**